Spec/RealtimeClientConnection.swift

```diff
@@ -2706,18 +2706,19 @@ class RealtimeClientConnection: QuickSpec {
                                         fail(error.message)
                                     }
                                     expect(channel.state).to(equal(ARTRealtimeChannelState.attached))
+                                    client.onDisconnected()
                                 }
                                 client.connection.once(.disconnected) { _ in
                                     client.connection.once(.connecting) { _ in
                                         client.connection.once(.connected) { _ in
                                             expect(client.connection.id).toNot(equal(connectionId))
-                                            channel.once(.attached) { _ in
+                                            channel.once(.attached) { stateChange in
+                                                expect(stateChange?.resumed).to(beFalse())
                                                 done()
                                             }
                                         }
                                     }
                                 }
-                                client.onDisconnected()
                             }
                         }
```

Comment:
I think the `client.connect()` should be called in the `waitUntil` closure. Just to be sure that all observers are set.

Reply:
@ricardopereira good point :) See https://github.com/ably/ably-ios/pull/726/commits/f0e2314b37160152a6a1af690fbc3167b0970831